libr/asm/arch/avr/format.c

```diff
@@ -156,10 +156,34 @@ static int formatDisassembledOperand(avrDisassembleContext *context, char *strOp
 			fOptions.addressFieldWidth, context->longAddress);
 		break;
 	case OPERAND_IO_REGISTER:
-		retVal = sprintf(strOperand, "%s%02x",
-			OPERAND_PREFIX_IO_REGISTER,
-			dInstruction.operands[operandNum]);
+	{
+		char *current_register = NULL;
+		int is_register_found = 0;
+		switch(dInstruction.operands[operandNum])
+		{
+		    case 0x3f:
+		        current_register = "sreg\0";
+		        break;
+		    case 0x04:
+		        current_register = "adcl\0";
+		        break;
+		    case 0x05:
+		        current_register = "adch\0";
+		        break;
+		    default:
+		        is_register_found = -1;
+		        retVal = sprintf(strOperand, "0x%x", dInstruction.operands[operandNum]);
+		        break;
+		    break;
+		}
+
+		if (is_register_found == 0) {
+			strcpy(strOperand, current_register);
+		}
+
+		retVal = strlen (strOperand);
```

Comment:
not sure why retval should be the length of the operand.. but if that's the case, youi know the length before hand

Reply:
There is a retVal to get the length of each sprintf. From the doc it returns the size of the string printed. Then I think so.

Reply:
feels like wrong design/unnecessary to me. btw im fine if you just introduce vulnerable code here, because the rest of the file is assuming very wrong buffer things, and that's out of the scope of this contribution. so you can remove all the calls to `sprintf` in a separate PR if in the mood.